src/compiler/mixin.cc

```diff
@@ -56,28 +83,53 @@ static void add_stubs(ir::Class* klass) {
       }
       auto original_parameters = method->parameters();
       ASSERT(original_parameters.length() == arity);
-      auto stub_parameters = ListBuilder<ir::Parameter*>::allocate(arity);
-      auto forward_arguments = ListBuilder<ir::Expression*>::allocate(arity);
-      for (int i = 0; i < arity; i++) {
-        auto original_parameter = original_parameters[i];
-        ASSERT(original_parameter->index() == i);
-        auto stub_parameter = _new ir::Parameter(original_parameter->name(),
-                                                 original_parameter->type(),
-                                                 original_parameter->is_block(),
-                                                 original_parameter->index(),
-                                                 original_parameter->has_default_value(),
-                                                 original_parameter->range());
-        stub_parameters[i] = stub_parameter;
-        forward_arguments[i] = _new ir::ReferenceLocal(stub_parameter, 0, range);
-      }
-
-      auto forward_call = _new ir::CallStatic(_new ir::ReferenceMethod(method, range),
-                                              shape.to_equivalent_call_shape(),
-                                              forward_arguments,
-                                              range);
-
+      auto stub_parameters = duplicate_parameters(original_parameters);
       ir::MethodInstance* stub;
-      if (method->is_IsInterfaceOrMixinStub()) {
+
+      if (method->is_FieldStub()) {
+        // Copy of what's happening in `resolver_method`.
+        auto range = method->range();
+        auto field_stub = method->as_FieldStub();
+        auto probe = new_fields.find(field_stub->field());
+        ASSERT(probe != new_fields.end());
+        auto new_field = probe->second;
+        ir::FieldStub* new_field_stub = _new ir::FieldStub(new_field,
+                                                           klass,
+                                                           field_stub->is_getter(),
+                                                           range);
+        new_field_stub->set_plain_shape(shape);
+        auto this_ref = _new ir::ReferenceLocal(stub_parameters[0], 0, range);
+        ir::Sequence* body;
+        if (field_stub->is_getter()) {
+          ASSERT(stub_parameters.length() == 1);
+          auto load = _new ir::FieldLoad(this_ref, new_field, range);
+          auto ret = _new ir::Return(load, false, range);
+          body = _new ir::Sequence(ListBuilder<ir::Expression*>::build(ret), range);
+        } else {
+          ASSERT(stub_parameters.length() == 2);
```

Comment:
No check for `field_stub->field()->is_final()`?

Reply:
Good catch.
I'm now forwarding the setter if the field is final.